### PR TITLE
edk2: migrate to bl2-el3 on HiKey/HiKey960

### PR DESCRIPTION
**Pull request by fboudra**
A new file "recovery.bin" is introduced and used in recovery mode.
A pseudo partition "loader" is created to store l-loader.bin file.
l-loader.bin need to flashed into the storage device using the fastboot
command.

* Update HiKey960 config: recovery.bin is used instead of l-loader.bin.
* Bump SRCREV to match latest code after bl2-el3 migration PRs have been
  merged.
* Switch to ATF master branch.

Signed-off-by: Fathi Boudra <fathi.boudra@linaro.org>

### CODE REVIEW

**Comment by fboudra** [comment]
@rsalveti FYI

File: recipes-bsp/uefi/edk2-hikey_git.bb [hunk 1]
```diff
@@ -21,7 +21,7 @@ SRCREV_atffastboot = "5b0d44c057bc0005965da990e8add72670810996"
 # knowledgeably, it is a hack...
 ###
 SRC_URI = "git://github.com/96boards-hikey/edk2.git;name=edk2;branch=testing/hikey960_v2.5 \
-           git://github.com/ARM-software/arm-trusted-firmware.git;name=atf;branch=integration;destsuffix=git/atf \
+           git://github.com/ARM-software/arm-trusted-firmware.git;name=atf;branch=master;destsuffix=git/atf \
```
Comment (hzhuang1) [inline recipes-bsp/uefi/edk2-hikey_git.bb:24]: The branch name is changed. Is there any reason that we choose master branch?

File: recipes-bsp/uefi/edk2-hikey960_git.bb [hunk 2]
```diff
@@ -6,15 +6,15 @@ DEPENDS_append = " dosfstools-native gptfdisk-native mtools-native virtual/faker
...
+SRCREV_openplatformpkg = "f686ca6485c73efd8a257e919f64b84772d331d5"
 SRCREV_uefitools = "42eac07beb4da42a182d2a87d6b2e928fc9a31cf"
-SRCREV_lloader = "79530f6d9b2668e2d3a76adadcc0053015937e82"
-SRCREV_toolsimageshikey960 = "ccb401f726346355e948ec776a411ad037bab4cc"
+SRCREV_lloader = "c1cbbf8ab824820b5c1769a1c80dd234c5b57ffc"
+SRCREV_toolsimageshikey960 = "9fd47e23d0afbfdcb4da507dc24f693a381eeec9"
 
 SRC_URI = "git://github.com/96boards-hikey/edk2.git;name=edk2;branch=testing/hikey960_v2.5 \
-           git://github.com/ARM-software/arm-trusted-firmware.git;name=atf;branch=integration;destsuffix=git/atf \
+           git://github.com/ARM-software/arm-trusted-firmware.git;name=atf;branch=master;destsuffix=git/atf \
```
Comment (hzhuang1) [inline recipes-bsp/uefi/edk2-hikey960_git.bb:17]: The branch name is changed. Is there any reason to choose master branch?

File: recipes-bsp/uefi/edk2-hikey_git.bb (hunk 1 above)
Comment (fboudra) [reply recipes-bsp/uefi/edk2-hikey_git.bb:24]: yes, master is preferred than integration. As of today, master was matching integration so I took th opportunity to switch.

**Review by koenkooi — APPROVED** [review]

**Pull request merged**